Data Science Fundamentals: Python |
[Table of Contents](../index.ipynb)
- - - 
<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Exercises](./04_mod_practice.ipynb)**

# Module 1: Practice Exercises

1. Accept the user's first and last name and print them in reverse order with a space between them

In [8]:
firstname = input("Input your First Name : ")
lastname = input("Input your Last Name : ")
print ("Hello " + lastname + " " + firstname)

Input your First Name : Tony
Input your Last Name : Mon
Hello Mon Tony


2. Accept an integer (n) and compute the value of n+nn+nnn

In [67]:
n=int(input("Enter a number n: "))
temp=str(n)
t1=temp+temp
t2=temp+temp+temp
comp=n+int(t1)+int(t2)
print("The value is:",comp)

Enter a number n: 4
The value is: 492


3. Ask the user "What country are you from?" then print the following statement: "I have heard that [input] is a beautiful country!"

In [7]:
country = input("What country are you from? : ")
print ("I have heard that " +  country  + " is a beautiful country!")

What country are you from? : USA
I have heard that USA is a beautiful country!


4. Write a for loop that multiplies the number 4 by all the integers from 0 to 10 and prints the following for each: "4 multiplied by [i] is [x]" 

In [34]:
    number = int(4)
    for count in range(0, 10):      
        print (number, 'x', count, '=', number * count)    

4 x 0 = 0
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36


5. Print a multiplication table that matches the following. Use two nested for loops to achieve this.

<img src="files/table.png">

In [62]:
rows=10
columns=10
for i in range(1, rows+1):
    for j in range(1, columns+1):
        c=i*j
        print("{:2d} ".format(c),end='')
    print('\n')

 1  2  3  4  5  6  7  8  9 10 

 2  4  6  8 10 12 14 16 18 20 

 3  6  9 12 15 18 21 24 27 30 

 4  8 12 16 20 24 28 32 36 40 

 5 10 15 20 25 30 35 40 45 50 

 6 12 18 24 30 36 42 48 54 60 

 7 14 21 28 35 42 49 56 63 70 

 8 16 24 32 40 48 56 64 72 80 

 9 18 27 36 45 54 63 72 81 90 

10 20 30 40 50 60 70 80 90 100 



6. Is a String Immutable in Python?

In [ ]:
Yes the value cannot be change

7. What is the output of x AND explain why that is the output for:

In [63]:
X = 36/4 * (5)*4 +2

In [64]:
x = 9*(5)*4 + 2

In [65]:
x = 45 * 4 + 2

In [66]:
x = 180 + 2

In [ ]:
x = 182

7. What is the output of the following code and what arithmetic operators is being used here? print(2%6)

In [68]:
print(2%6)

2


2, % is Modulus

8. What is the output of the following code and what arithmetic operators are used here? print(2 * 3 ** 3 * 4)

In [69]:
print(2 * 3 ** 3 * 4)

216


216 ** is Exponentiation  and  * is Multiplication

9. What is a text editor?

A text editor is a program that lets a user enter, change, store, and usually print text 

10. What is python?

Python is an interpreted, object-oriented, high-level programming language with dynamic semantics.

11. What is jupyter notebook, what type of python environment is it, and what alternatives are there to jupyter notebook?

Jupyter notebook is an interactive development environment that allows editing and running notebook documents via a web browser. There many alternatives like PyCharm, Apache Zeppelin, and RStudio.  

12. What is the output of the following addition (+) operator, and why does this code chunk execute this way?

In [72]:
a = [10, 20]
b = a
b += [30, 40]
print(a)
print(b)

[10, 20, 30, 40]
[10, 20, 30, 40]


In [ ]:
unchanged addition since a = 10 20, b = a, b += 30 40 print(a) = 10 20 30 40 same for b

13. What is the output of the following Python code

In [81]:
x = 10
y = 50

if (x ** 2 > 100 and y < 100):
    print(x, y)

x = 2500 y = 50 

<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Practice](./04_mod_practice.ipynb)**
<br>
[top](#)

- - -

Copyright © 2020 Qualex Consulting Services Incorporated.